File: website/open-build.js
(function () {
  const refused = ["copy gpt", "copy claude", "distill gpt", "distill claude", "proprietary weights", "closed weights"];
  const slug = /^[A-Za-z0-9][A-Za-z0-9._-]{0,80}\/[A-Za-z0-9][A-Za-z0-9._-]{0,80}$/;
  const secret = /ghp_|github_pat_|sk-|hf_|AKIA/;
  const status = document.getElementById("build-status");

  document.getElementById("build-form").addEventListener("submit", function (event) {
    event.preventDefault();
    const source = document.getElementById("source").value;
    const name = document.getElementById("name").value.trim();
    const goal = document.getElementById("goal").value;
    const sample = document.getElementById("sample").value;
    const blob = (source + " " + name + " " + goal).toLowerCase().replace(/[-_\/]/g, " ");
    const link = document.getElementById("build-link");
    function stop(text) { status.textContent = text; link.hidden = true; }
    if (refused.some(function (phrase) { return blob.indexOf(phrase) !== -1; })) return stop("Closed weights are not a starting point.");
    if (!slug.test(name)) return stop("Use owner/name.");
    if (secret.test(sample)) return stop("A key-shaped string is in the pasted code. Remove it. The code was not run.");
    if (!document.getElementById("license-ok").checked || !document.getElementById("owns").checked || !document.getElementById("hidden").checked) {
      return stop("You need a license that allows training, data you own, and a hidden test.");
    }
    link.hidden = false;
    link.href = (source === "github" ? "https://github.com/" : "https://huggingface.co/") + name;
    link.textContent = "Open this open-source starting point";
    status.textContent = "Plan accepted. Read the license, then train on a machine you control. This page did not build a frontier model, did not run the pasted code, and did not score anyone else's benchmark.";
  });
})();

File: buddy/learning/open_build.py
#!/usr/bin/env python3
"""Plan a model from open source. Do not train it here."""
from __future__ import annotations

import json
import re

SLUG = re.compile(r"^[A-Za-z0-9][A-Za-z0-9._-]{0,80}/[A-Za-z0-9][A-Za-z0-9._-]{0,80}$")
REFUSED = (
    "copy gpt",
    "copy claude",
    "distill gpt",
    "distill claude",
    "proprietary weights",
    "closed weights",
)


def plan(source: str, name: str, goal: str, license_ok: bool, owns_data: bool, hidden_test: bool) -> dict:
    text = f"{source} {name} {goal}".lower().replace("-", " ").replace("_", " ").replace("/", " ")
    for phrase in REFUSED:
        if phrase in text:
            return {"accepted": False, "trains_here": False, "frontier_built": False, "reason": "Closed weights are not a starting point."}
    if source not in {"github", "huggingface"}:
        return {"accepted": False, "trains_here": False, "frontier_built": False, "reason": "Use a public GitHub repository or a public Hugging Face model."}
    if not SLUG.match((name or "").strip()):
        return {"accepted": False, "trains_here": False, "frontier_built": False, "reason": "Use owner/name."}
    if not (license_ok and owns_data and hidden_test):
        return {
            "accepted": False,
            "trains_here": False,
            "frontier_built": False,
            "reason": "You need a license that allows training, data you own, and a hidden test.",
        }
    clean = name.strip()
    url = f"https://github.com/{clean}" if source == "github" else f"https://huggingface.co/{clean}"
    return {
        "accepted": True,
        "trains_here": False,
        "frontier_built": False,
        "benchmark": "Our own gates only. No frontier model was scored.",
        "opens": url,
        "reason": "Plan accepted. Read the license, then train on a machine you control. This page did not build the model.",
    }


if __name__ == "__main__":
    bad = plan("github", "org/repo", "distill Claude", True, True, True)
    assert bad["accepted"] is False and bad["frontier_built"] is False
    missing = plan("huggingface", "org/model", "practice summaries", False, True, True)
    assert missing["accepted"] is False
    good = plan("github", "org/open-code", "train on our notes", True, True, True)
    assert good["accepted"] is True and good["trains_here"] is False and good["frontier_built"] is False
    assert plan("secret", "org/model", "notes", True, True, True)["accepted"] is False
    print(json.dumps(good))
